aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+import os
+import sys
+import netCDF4 as nc4
+import numpy as np
+import json
+from datetime import datetime
+import argparse
+
+import xarray as xr
+
+import aggregated_timeseries as TStools
+
+TEMPLATE_JSON = 'velocity_aggregated_timeseries_template.json'
+
+
+def sort_files(files_to_agg):
+    """
+    sort list of files according to deployment date
+    requires netcdf4 as nc4, dateutil.parser as parse
+    :param files_to_agg: List of files to sort
+    :return: sorted list of files
+    """
+
+    time_start = []
+    for file in files_to_agg:
+        with nc4.Dataset(file, 'r') as ds:
+            time_start.append(np.datetime64(ds.time_deployment_start))
+    tuples = sorted(zip(time_start, files_to_agg))
+    return [t[1] for t in tuples]
+
+
+def check_file(nc, site_code):
+    """
+    Return list of errors found in the file:
+    Variables of interest are present
+    TIME. LATITUDE, LONGITUDE,  is present
+    NOMINAL_DEPTH is not present as variable or attribute
+    file_version is not FV01
+    if LATITUDE or LONIGUTDE dimension has length >1
+
+    :param file: name of the netcdf file
+    :param nc: xarray dataset
+    :param VoI: string. Variable of Interest
+    :param site_code: code of the mooring site
+    :return: dictionary with the file name and list of failed tests
+    """
+
+    attributes = list(nc.attrs)
+    file_variables = list(nc.variables)
+    allowed_dimensions = ['TIME', 'LATITUDE', 'LONGITUDE', 'HEIGHT_ABOVE_SENSOR']
+    required_variables = ['UCUR', 'VCUR', 'WCUR']
+    error_list = []
+
+    nc_site_code = nc.site_code
+    if nc_site_code != site_code:
+        error_list.append('Wrong site_code: ' + nc_site_code)
+
+    nc_file_version = nc.file_version
+    if 'Level 1' not in nc_file_version:
+        error_list.append('Wrong file version: ' + nc_file_version)
+
+    for variable in required_variables:
+        if variable not in file_variables:
+            error_list.append(variable + ' variable missing')
+        else:
+            VoIdimensions = list(nc[variable].dims)
+            if 'TIME' not in VoIdimensions:
+                error_list.append('TIME is not a dimension for ' + variable)
+            if 'LATITUDE' in VoIdimensions and len(nc.LATITUDE) > 1:
+                error_list.append('more than one LATITUDE for ' + variable)
+            if 'LONGITUDE' in VoIdimensions and len(nc.LONGITUDE) > 1:
+                error_list.append('more than one LONGITUDE for ' + variable)
+            for dim in VoIdimensions:
+                if dim not in allowed_dimensions:
+                    error_list.append('not allowed dimension: ' + dim)
+
+    if 'NOMINAL_DEPTH' not in file_variables and 'instrument_nominal_depth' not in attributes:
+        error_list.append('no NOMINAL_DEPTH')
+
+    return error_list
+
+
+def get_nvalues(nc):
```

Comment:
nvalues means nothing here, hence intent is not clear. 

e.g. get_number_of_vertical_cells is much better - I don't even need to read the docstring.

Reply:
Actually,why not just return the HEIGHT and the TIME variables? The intent will be clearer in the function call - The name would be "get_dim_len" or alike.

Reply:
I've renamed the function and variables for clarity. The objective is to return the number of values that result in the flattening of the grid

Reply:
see comment below about this function - it's not necessary.